addons/account/tests/test_account_invoice_report.py

```diff
@@ -117,3 +117,57 @@ def test_invoice_report_multiple_types(self):
             [20, -20, -1],
             [600, -600, -1],
         ])
+
+    def test_avg_price_calculation(self):
+        """
+        Check that the average is correctly calculated based on the total price and quantity:
+            3 lines:
+                - 10 units * 10$
+                -  5 units *  5$
+                - 20 units *  2$
+            Total quantity: 35
+            Total price: 165$
+            Average: 165 / 35 = 4.71
+        """
+        product = self.product_a.copy()
+        invoice = self.env["account.move"].create({
+            'move_type': 'out_invoice',
+                'partner_id': self.partner_a.id,
+                'invoice_date': fields.Date.from_string('2016-01-01'),
+                'currency_id': self.env.company.currency_id.id,
+                'invoice_line_ids': [
+                    (0, None, {
+                        'product_id': product.id,
+                        'quantity': 10,
+                        'price_unit': 10,
+                    }),
+                    (0, None, {
+                        'product_id': product.id,
+                        'quantity': 5,
+                        'price_unit': 5,
+                    }),
+                    (0, None, {
+                        'product_id': product.id,
+                        'quantity': 20,
+                        'price_unit': 2,
+                    }),
+                ]
+        })
+        invoice.action_post()
+
+        report = self.env['account.invoice.report'].read_group(
+            [('product_id', '=', product.id)],
+            ['price_subtotal', 'quantity', 'price_average:avg'],
+            [],
+        )
+        self.assertEqual(report[0]['quantity'], 35)
+        self.assertEqual(report[0]['price_subtotal'], 165)
+        self.assertEqual(round(report[0]['price_average'], 2), 4.71)
+
+        # ensure that it works with only 'price_average:avg' in fields
+        report = self.env['account.invoice.report'].read_group(
+            [('product_id', '=', product.id)],
+            ['price_average:avg'],
+            [],
+        )
+        self.assertEqual(round(report[0]['price_average'], 2), 4.71)
```

Comment:
> Then why not remove price_subtotal and quantity from the test ?

@rugo-odoo do you prefer that I only check without `price_subtotal` and `quantity` or this is ok?

Reply:
Yep, this is okay.